### DISCUSSION TITLE
Unable to connect to Redis over a unix socket

### DISCUSSION THREAD

**Discussion by Zepmann** [authelia/authelia#6427] [Troubleshooting] [answered]
Running the current git version ([47fe5c0](https://github.com/authelia/authelia/commit/47fe5c01bc71801f3b2f050cc1bc949f0b0ba4fd)) on Arch Linux using AUR package [authelia-git](https://aur.archlinux.org/packages/authelia-git).

Relevant Authelia configuration:
```
session:
  redis:   
    host: '/run/redis/redis-server.sock'
```

Redis is running and listening on the unix socket:
```
$ ls -al /run/redis
total 0
drwxr-xr-x  2 redis redis  60 Dec 16 23:17 .
drwxr-xr-x 24 root  root  820 Dec 16 23:17 ..
srwxrwx---  1 redis redis   0 Dec 16 23:17 redis-server.sock
```

Authelia has access to the socket. The same configuration was tested using v4.37.5.

When starting Authelia, the following error occurs and Authelia quits:
```
level=info msg="Authelia untagged-v4.37.5 (master, 47fe5c0) is starting"
level=info msg="Log severity set to info"
level=fatal msg="redis connection error: dial tcp: lookup /run/redis/redis-server.sock: no such host" stack="github.com/authelia/authelia/v4/internal/commands/context.go:147 (*CmdCtx).LoadProviders\ngithub.com/authelia/authelia/v4/internal/commands/root.go:71     (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\n./main.go:10                                                     main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1650                                         goexit"
```

The [documentation states](https://github.com/authelia/authelia/blob/master/docs/content/en/configuration/session/redis.md#host):
> The redis host **or unix socket path**. If utilising an IPv6 literal address it must be enclosed by square brackets and quoted:

The [example configuration states](https://github.com/authelia/authelia/blob/master/config.template.yml#L766C1-L770C40):
```
  # redis:
    # host: '127.0.0.1'
    # port: 6379
    ## Use a unix socket instead
    # host: '/var/run/redis/redis.sock'
```

The same Redis installation works fine in Authelia v4.37.5 using configuration:
```
session:
  redis:
    host: /run/redis/redis-server.sock
```

I did try the same configuration (without quotes) in the tested git version, but got the same error.

Please advise on how to solve this issue.

Workaround:
Have Redis listen on a TCP socket and connect to it that way (e.g. '127.0.0.1:6379' or '[::1:6379]').

**Comment by james-d-elliott** [comment, marked as answer]
Fixed in f93a0b5bf9d2a1d29a08c7b8c699647ca627bdb8
